peer/pendingheap/heap_test.go

```diff
@@ -269,16 +261,7 @@ func TestReleaseLockWithStaleSubscriber(t *testing.T) {
 	ph.Remove(p1, p1, subscriber)
 
 	// simulate transport (eg HTTP) ending a call to a peer
-	ph.notifyStatusChanged(subscriber)
-
-	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*100)
-
-	// all subsequent calls should succeed if we're not holding onto the lock
-	go func() {
-		ph.Choose(ctx, &transport.Request{})
-		cancel()
-	}()
+	ph.updatePendingRequestCount(subscriber, 0)
 
-	<-ctx.Done()
-	assert.Equal(t, context.Canceled, ctx.Err(), "expected context to be canceled")
+	ph.Choose(&transport.Request{})
```

Comment:
Removing the context/cancelation logic means that a failing test would hang here with no description
Can we put something similar back or at least add a comment here indicating what's wrong if it failed?

Reply:
Sure.